hazardous/data/_competing_weibull.py

```diff
@@ -84,21 +167,121 @@ def make_synthetic_competing_weibull(
     event_durations = np.asarray(event_durations)
     duration_argmin = np.argmin(event_durations, axis=0)
     X = pd.concat(all_features, axis=1)
+    return X, event_durations, duration_argmin
+
+
+def make_complex_features_with_sparse_matrix(
+    n_events=3,
+    n_samples=3_000,
+    base_scale=1_000,
+    shape_ranges=DEFAULT_SHAPE_RANGES,
+    scale_ranges=DEFAULT_SCALE_RANGES,
+    n_features=10,
+    features_rate=0.3,
+    degree_interaction=2,
+    random_state=0,
+):
+    rng = np.random.RandomState(random_state)
+    # Create features given to the model as X and then creating the interactions
+    df_features = pd.DataFrame(rng.randn(n_samples, n_features))
+    df_features.columns = [f"feature_{i}" for i in range(n_features)]
+
+    df_shape_scale_star = compute_shape_and_scale(
+        df_features,
+        features_rate,
+        n_events,
+        degree_interaction,
+        shape_ranges,
+        scale_ranges,
+        random_state,
+    )
+    # Throw durations from a weibull distribution with scale and shape as the parameters
+    event_durations = []
+    for event in range(1, n_events + 1):
+        shape = df_shape_scale_star[f"shape_{event}"]
+        scale = df_shape_scale_star[f"scale_{event}"]
+        durations = weibull_min.rvs(shape, scale=scale * base_scale, random_state=rng)
+        event_durations.append(durations)
+
+    # Creating the target tabular
+    event_durations = np.asarray(event_durations)
+    duration_argmin = np.argmin(event_durations, axis=0)
+    return df_features, event_durations, duration_argmin
+
+
+def make_synthetic_competing_weibull(
+    n_events=3,
+    n_samples=3_000,
+    base_scale=1_000,
+    n_features=10,
+    features_rate=0.3,
+    degree_interaction=2,
+    independent=True,
```

Comment:
To be more specific?
```suggestion
    independent_censoring=True,
```